urldeb33fd2fa_friedemann_distripedia_tgz-pJ8-org_json_TestJ8/src/grinder/controller/CtrlServerThread.java

```diff
@@ -0,0 +1,359 @@
+package grinder.controller;
+
+import grinder.client.ClientCommand;
+import grinder.client.ClientState;
+import grinder.util.GrinderSettings;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.io.OptionalDataException;
+import java.net.InetAddress;
+import java.net.Socket;
+import java.util.UUID;
+
+import javax.swing.ProgressMonitor;
+
+/**
+ * The Class CtrlServerThread handles the communication with the connected
+ * client - one thread each connection.
+ *
+ * @author Friedemann
+ */
+public class CtrlServerThread extends Thread {
+
+	// reference to papa controller
+	private final Controller ctrl;
+	// reference to papa server
+	private final CtrlServer server;
+
+	/** The client socket. */
+	private final Socket clientSocket;
+
+	/** Thread name. */
+	private final String name;
+
+	private ObjectOutputStream oos;
+	private ObjectInputStream ois;
+
+	// client information
+	private UUID clientID;
+	private InetAddress clientIP;
+	private ClientState clientState;
+	private String deployedTicketID = "";
+
+	/** Controls the while loop within run() */
+	private boolean serverThreadListening = true;
+
+	/** The progress monitor. */
+	private ProgressMonitor pMonitor;
+
+	/**
+	 * Instantiates a new server thread.
+	 *
+	 * @param ctrl reference to controller
+	 * @param server reference to controlserver thread
+	 * @param name thread name
+	 * @param socket bound client socket
+	 */
+	public CtrlServerThread(final Controller ctrl, final CtrlServer server,
+			final String name, final Socket socket) {
+		super(name);
+
+		this.ctrl = ctrl;
+		this.server = server;
+		clientSocket = socket;
+		this.name = name;
+
+		ctrl.cgui.logMessage("<" + name + "> server connection thread initialized");
+	}
+
+	/**
+	 * Write to object output stream.
+	 *
+	 * @param obj the obj
+	 */
+	public void writeToOutputStream(final Object objOut) {
+		if (oos != null)
+		{
+			try
+			{
+				oos.writeObject(objOut);
+				oos.flush();
+
+			} catch (final IOException e)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)